icu4j/main/core/src/test/java/com/ibm/icu/dev/test/format/JavaTimeConvertersTest.java

```diff
@@ -0,0 +1,156 @@
+import java.time.LocalTime;
+import java.time.Month;
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.chrono.HijrahDate;
+import java.time.chrono.JapaneseDate;
+import java.time.chrono.JapaneseEra;
+import java.time.chrono.MinguoDate;
+import java.time.chrono.ThaiBuddhistDate;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import com.ibm.icu.impl.JavaTimeConverters;
+import com.ibm.icu.util.Calendar;
+import com.ibm.icu.util.GregorianCalendar;
+import com.ibm.icu.util.TimeZone;
+
+/* This class tests the raw conversion, java.time classes to an ICU Calendar. */
+@RunWith(JUnit4.class)
+public class JavaTimeConvertersTest {
+
+    /*
+     * Fields that we expect in the calendar when formatting dates.
+     *
+     * A LocalDate object will not have hour, minutes, seconds, etc.
+     * So when we convert it to a Calendar the result can't be directly compared
+     * to the expected Calendar because some fields are different.
+     *
+     * Think of this field list as a "mask" we use when we compare a calendar
+     * from conversion with the expected Calendar.
+     */
+    private final static int[] DATE_ONLY_FIELDS = {
+            Calendar.DAY_OF_MONTH, Calendar.MONTH, Calendar.YEAR,
+            Calendar.DAY_OF_WEEK, Calendar.DAY_OF_YEAR, Calendar.ERA,
+            Calendar.DAY_OF_WEEK_IN_MONTH, Calendar.DOW_LOCAL,
+            Calendar.WEEK_OF_MONTH, Calendar.WEEK_OF_YEAR, Calendar.EXTENDED_YEAR
+    };
+
+    // Fields that we expect in the calendar when formatting time
+    private final static int[] TIME_ONLY_FIELDS = {
+            Calendar.HOUR_OF_DAY, Calendar.MINUTE, Calendar.SECOND, Calendar.MILLISECOND,
+            Calendar.AM_PM, Calendar.HOUR_OF_DAY, Calendar.MILLISECONDS_IN_DAY
+    };
+
+    // Make it easier to build all kind of temporal objects
+    final static LocalDateTime LOCAL_DATE_TIME = LocalDateTime.of(2018, Month.SEPTEMBER, 23,
+        19, 42, 57, /*nanoseconds*/ 123_000_000);
+
+    final static String TIME_ZONE_ID = "Europe/Paris";
+
+    // Match the fields in the LOCAL_DATE_TIME above
+    final static Calendar EXPECTED_CALENDAR = new GregorianCalendar(2018, Calendar.SEPTEMBER,
+            23, 19, 42, 57);
+    static {
+        EXPECTED_CALENDAR.setTimeZone(TimeZone.getTimeZone(TIME_ZONE_ID));
+        EXPECTED_CALENDAR.setTimeInMillis(EXPECTED_CALENDAR.getTimeInMillis() + 123);
+    }
+
+    @Test
+    public void testDateOnly() {
+        LocalDate ld = LOCAL_DATE_TIME.toLocalDate();
+        Calendar calendar = JavaTimeConverters.temporalToCalendar(ld);
+        assertCalendarsEquals(EXPECTED_CALENDAR, calendar, DATE_ONLY_FIELDS);
+
+        HijrahDate hd = HijrahDate.of(1440, 1, 13);
+        calendar = JavaTimeConverters.temporalToCalendar(hd);
+        assertCalendarsEquals(EXPECTED_CALENDAR, calendar, DATE_ONLY_FIELDS);
```

Comment:
I's interesting that the conversion from `java.time` to ICU `Calendar` doesn't preserve the calendrical system of the input when providing the output. (The `EXPECTED_CALENDAR` expected value is a Gregorian calendar date).

Do users already know not to assume that the output calendar system will match, or is it something worth documenting somewhere?

Reply:
It's complicated :-)
It is not like there is one clear winner.

See here:
> _When the calendar type is different from the internal calendar held by this DateFormat instance, the date and the time zone will be inherited from the input calendar, but other calendar field values will be calculated by the internal calendar._
>
> https://unicode-org.github.io/icu-docs/apidoc/released/icu4j/com/ibm/icu/text/DateFormat.html#format-com.ibm.icu.util.Calendar-java.lang.StringBuffer-java.text.FieldPosition-